src/nrrd/write.c

```diff
@@ -1033,10 +1033,18 @@ nrrdSave(const char *filename, const Nrrd *nrrd, NrrdIoState *nio) {
     nio->detachedHeader = AIR_FALSE;
   }
 
-  if (!( file = airFopen(filename, stdout, "wb") )) {
-    biffAddf(NRRD, "%s: couldn't fopen(\"%s\",\"wb\"): %s",
-             me, filename, strerror(errno));
-    airMopError(mop); return 1;
+  if (!nio->keepNrrdDataFileOpen) {
+    if (!( file = airFopen(filename, stdout, "wb") )) {
+      biffAddf(NRRD, "%s: couldn't fopen(\"%s\",\"wb\"): %s",
+               me, filename, strerror(errno));
+      airMopError(mop); return 1;
+    }
+  } else {
+    if (!( file = airFopen(filename, stdout, "ab") )) {
+      biffAddf(NRRD, "%s: couldn't fopen(\"%s\",\"ab\"): %s",
+               me, filename, strerror(errno));
+      airMopError(mop); return 1;
```

Comment:
It would be nice if we could check that there is no gap between the end of last written chunk and start of current chunk; and no overlap between already written chunks and current chunk. Just logging an error and returning with failure would work (there is no need to implement leaving gaps or overwriting).

Reply:
ok